src/accelerate/state.py

```diff
@@ -22,6 +22,7 @@
 from contextlib import contextmanager
 from functools import partial
 from typing import Any, Callable, Optional
+from datasets import Dataset
```

Comment:
We cannot have this. Let's do a check for `is_datasets_available()` and potentially add it into the import utils file.